src/components/ui/drawers/NameEditionDrawer.tsx

```diff
@@ -104,34 +110,57 @@ export const NameEditionDrawer = ({
 
   // TODO: Implement the onSubmit click functionality
   const onSubmit: SubmitHandler<FormSchema> = async (data) => {
-    setIsLoading(true);
-    setOpen(true);
-
     // Submit the form data
     await new Promise((resolve) => setTimeout(resolve, 1000));
 
-    // Update the name based on the id
-    if (name) {
-      setIsLoading(false);
+    try {
+      if (isSharehouse) {
+        const resSharehouse = await api.sharehouse[':shareHouseId'].$patch({
+          param: {
+            shareHouseId: id!,
+          },
+          json: {
+            name: data.name,
+          },
+        });
+        const sharehouseData = await resSharehouse.json();
+        if ('error' in sharehouseData) {
+          throw new Error(sharehouseData.error);
+        }
+      } else {
+        const resCategory = await api.category[':categoryId'].$patch({
+          param: {
+            categoryId: id!,
```

Comment:
The non-null assertion could lead to unintended behavior later in development. Could you either add a comment on the line to remind us to fix it later or add an if statement to ensure the submission only happens if the ID is not null?